include/grpc++/server_context.h

```diff
@@ -125,6 +125,11 @@ class ServerContext {
 
   const struct census_context* census_context() const;
 
+  // Async only. Has to be called before the rpc starts.
+  // Returns the tag in completion queue when the rpc finishes.
+  // IsCancelled() can then be called to check whether the rpc was cancelled.
+  void AsyncNotifyWhenDone(void* tag) { async_notify_when_done_tag_ = tag; }
```

Comment:
It's legal to have a NULL tag... we should probably also have a `bool have_tag_` field here.